src/query/service/src/interpreters/interpreter_table_create.rs

```diff
@@ -283,3 +286,17 @@ pub static CREATE_TABLE_OPTIONS: Lazy<HashSet<&'static str>> = Lazy::new(|| {
 pub fn is_valid_create_opt<S: AsRef<str>>(opt_key: S) -> bool {
     CREATE_TABLE_OPTIONS.contains(opt_key.as_ref().to_lowercase().as_str())
 }
+
+pub fn is_valid_block_per_segment(options: &BTreeMap<String, String>) -> Result<()> {
+    // check block_per_segment is not over 1000.
+    if let Some(value) = options.get(FUSE_OPT_KEY_BLOCK_PER_SEGMENT) {
+        let blocks_per_segment = value.parse::<u64>()?;
+        if blocks_per_segment > 1000 {
+            error!("invalid block_per_segment option, can't be over 1000");
+            return Err(ErrorCode::TableOptionInvalid(format!(
+                "invalid block_per_segment option, can't be over 1000"
```

Comment:
shall we introduce an error string, and use it both in the log message and error message?

Reply:
yes